### DISCUSSION TITLE
feature: parallel quest objectives/tasks

### DISCUSSION THREAD

**Discussion by froggytrav** [AscensionGameDev/Intersect-Engine#1536] [Ideas]
**Is your feature request related to a problem? Please describe.**
I am always frustrated when designing quests and you have to create one quest per NPC to kill, or item to gat her.
For example, say I am designing and a dungeon, and there will be quests to eliminate the monsters of said dungeon. Instead of having to create the same amount of quests as there are types of monsters in the dungeon, you should be able to allow players to complete specific tasks in any order

**Describe the solution you'd like**
What I want to happen is a (the way to go about it is up to the devs i think?) a setting where you can group up certain tasks (regardless of kill/gather), so it would look something like this:
*Quest starts*
NPC tasks player with eliminating 2 types of monsters, A & B.
--Task 1: Eliminate 8 Monster A
--Task 2: Eliminate 8 Monster B
--Task 3: Gather 6 Shoes
Task 1, 2 & 3 completed --> return to NPC

So the solution here would be to somehow group up task 1, 2 & 3 so they can be completed and worked on in any order. The current workaround is to create a quests for each task.

**Additional context**
Context to this is that this is a pretty much basic MMO/RPG feature in just about any game. It would save space and reduce clutter when creating quests/tasks etc

**Comment by Arufonsu** [comment]
Could you edit this feature request (or enhancement?) by following the proper format?


```
<!--- Follow the template below. Feature requests that do not follow the following template will be closed. -->
<!--- Also, please note that Intersect is designed as a base or platform to design 2d mmorpgs, niche features, or features that we feel won't apply to most games will not be considered. -->

**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]

**Describe the solution you'd like**
A clear and concise description of what you want to happen.

**Additional context**
Add any other context or screenshots about the feature request here.
```

**Comment by Arufonsu** [comment]
Considering the amount of effort and rewritting required for this to happen, and according to the discussion held with collaborators, im re-labeling this as a feature request. The request itself is quite reasonable and If anyone is manages to tackle this one down, we are open to review the PR.
